index.rb

```diff
@@ -0,0 +1,139 @@
+
+    res = @result_selection.filter { |obj| obj['model'] == sanitize_text(model_to_find) }
+    @result_selection = res.empty? ? @app_errors.push("Can't find by model: #{model_to_find}") : res
+  end
+
+  def get_by_year_from(year_to_find)
+    return if year_to_find.length.zero?
+
+    res = @result_selection.filter { |obj| obj['year'] >= sanitize_text(year_to_find).to_i }
+    @result_selection = res.empty? ? @app_errors.push("Can't find by year: #{year_to_find}") : res
+  end
+
+  def get_by_year_to(year_to)
+    return if year_to.length.zero?
+
+    res = @result_selection.filter { |obj| obj['year'] <= sanitize_text(year_to).to_i }
+    @result_selection = res.empty? ? @app_errors.push("Can't find by year: #{year_to}") : res
+  end
+
+  def get_by_price_from(price)
+    return if price.length.zero?
+
+    res = @result_selection.filter { |obj| obj['price'] >= sanitize_text(price).to_i }
+    @result_selection = res.empty? ? @app_errors.push("Can't find by price_from: #{price}") : res
+  end
+
+  def get_by_price_to(price)
+    return if price.length.zero?
+
+    res = @result_selection.filter { |obj| obj['price'] <= sanitize_text(price).to_i }
+    @result_selection = res.empty? ? @app_errors.push("Can't find by price_to: #{price}") : res
+  end
+
+  def sort_by_criterias(direction, option = 'desc')
+    if 'price' == option
+      @result_selection = @result_selection.sort { |one, two| one[option] <=> two[option] }
+    elsif 'date_added' == option
+      @result_selection = @result_selection.sort { |one, two| one[option] <=> two[option] }
+    end
+    @result_selection = direction == 'asc' ? @result_selection : @result_selection.reverse
+  end
+
+  def init
+    puts 'Please select search rules'
+
+    puts 'Please choose make:'
+    @make = gets.chomp
+
+    puts 'Please choose model:'
+    @model = gets.chomp
+
+    puts 'Please choose year_from:'
+    @year_from = gets.chomp
+
+    puts 'Please choose year_to:'
+    @year_to = gets.chomp
+
+    puts 'Please choose price_from:'
+    @price_from = gets.chomp
+
+    puts 'Please choose price_to:'
+    @price_to = gets.chomp
+
+    puts 'Please choose sort option (date_added|price):'
+    @date_or_price_sort = gets.chomp
+
+    puts 'Please choose sort direction(desc|asc):'
+    @asc_desc_sort = gets.chomp
+
+    fetch_data
+  end
+
+  def fetch_data
+    unless @make && @model && @year_from && @year_to && @price_from && price_to && @asc_desc_sort && @date_or_price_sort
+      return
+    end
+
+    get_by_make(@make)
+    get_by_model(@model)
+    get_by_year_from(@year_from)
+    get_by_year_to(@year_to)
+    get_by_price_from(@price_from)
+    get_by_price_to(@price_to)
+    sort_by_criterias(@asc_desc_sort, @date_or_price_sort)
+    display_result
+  end
+
+  def display_result
+    if !@app_errors.empty?
+      puts 'Errors', @app_errors
+    else
+      puts '=================================='
+      puts 'Display results', @result_selection
```

Comment:
![Знімок екрана з 2022-11-07 12-24-31](https://user-images.githubusercontent.com/22034821/200287252-3bf52fe2-1132-48de-a55f-c04f7b5a10de.png)
In task we have an example how it can be displayed:
![Знімок екрана з 2022-11-07 12-27-20](https://user-images.githubusercontent.com/22034821/200287829-e38c192d-2f0c-4f2d-9a03-72ca0bf4c553.png)
Can you prettify result in console. for example:       
```suggestion
      puts 'Display results'
      @result_selection.each do |car|
        car.each do |key, value|
          puts "#{key.capitalize}: #{value}"
        end
        puts
      end
```